retrieval_service/datastore/providers/spanner_gsql_test.py

```diff
@@ -0,0 +1,640 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime
+from ipaddress import IPv4Address
+from typing import Any, AsyncGenerator, List, Optional
+
+import pytest
+import pytest_asyncio
+from csv_diff import compare, load_csv  # type: ignore
+from google.cloud import spanner  # type: ignore
+from google.cloud.spanner_v1 import JsonObject, param_types
+from google.cloud.spanner_v1.database import Database
+from google.cloud.spanner_v1.instance import Instance
+from google.oauth2 import service_account
+
+import models
+
+from .. import datastore
+from . import spanner_gsql
+from .test_data import (
+    amenities_query_embedding1,
+    amenities_query_embedding2,
+    foobar_query_embedding,
+    policies_query_embedding1,
+    policies_query_embedding2,
+)
+from .utils import get_env_var
+
+pytestmark = pytest.mark.asyncio(scope="module")
+
+
+@pytest.fixture(scope="module")
+def db_project() -> str:
+    return get_env_var("DB_PROJECT", "Google Cloud Project")
+
+
+@pytest.fixture(scope="module")
+def db_instance() -> str:
+    return get_env_var("DB_INSTANCE", "Spanner Instance")
+
+
+@pytest.fixture(scope="module")
+def db_name() -> str:
+    return get_env_var("DB_NAME", "Spanner Database")
+
+
+@pytest.fixture(scope="module")
+def service_accounts_file_path() -> str:
```

Comment:
The test failed with this since there's no key file provided in the integration test. We can probably just remove this? I made it optional for spanner_gsql previously. It should be able to use cloud build's ADC

Reply:
removed it, it's working locally. Let's see if it works.